runtime/src/main/java/com/vaadin/quarkus/context/BeanProvider.java

```diff
@@ -0,0 +1,597 @@
+     * case!
+     * </p>
+     *
+     * @param type
+     *            the type of the bean in question
+     * @param optional
+     *            if <code>true</code> it will return an empty list if no bean
+     *            could be found or created. Otherwise it will throw an
+     *            {@code IllegalStateException}
+     * @param includeDefaultScopedBeans
+     *            specifies if dependent scoped beans should be included in the
+     *            result
+     * @param <T>
+     *            target type
+     *
+     * @return the resolved list of Contextual Reference or an empty-list if
+     *         optional is true
+     */
+    public static <T> List<T> getContextualReferences(Class<T> type,
+            boolean optional, boolean includeDefaultScopedBeans) {
+        BeanManager beanManager = getBeanManager();
+
+        Set<Bean<T>> beans = getBeanDefinitions(type, optional,
+                includeDefaultScopedBeans, beanManager);
+
+        List<T> result = new ArrayList<T>(beans.size());
+
+        for (Bean<?> bean : beans) {
+            // noinspection unchecked
+            result.add(getContextualReference(type, beanManager, bean));
+        }
+        return result;
+    }
+
+    /*
+     * public static <T> DependentProvider<T> getDependent(Class<T> type,
+     * Annotation... qualifiers) { BeanManager beanManager = getBeanManager();
+     * return getDependent(beanManager, type, qualifiers); }
+     * 
+     * public static <T> DependentProvider<T> getDependent(BeanManager
+     * beanManager, Class<T> type, Annotation... qualifiers) { Set<Bean<?>>
+     * beans = beanManager.getBeans(type, qualifiers);
+     * 
+     * @SuppressWarnings("unchecked") Bean<T> bean = (Bean<T>)
+     * beanManager.resolve(beans); return createDependentProvider(beanManager,
+     * type, bean); }
+     * 
+     * public static <T> DependentProvider<T> getDependent(String name) {
+     * BeanManager beanManager = getBeanManager(); return
+     * getDependent(beanManager, name); }
+     * 
+     * public static <T> DependentProvider<T> getDependent(BeanManager
+     * beanManager, String name) { Set<Bean<?>> beans =
+     * beanManager.getBeans(name);
+     * 
+     * @SuppressWarnings("unchecked") Bean<T> bean = (Bean<T>)
+     * beanManager.resolve(beans);
+     * 
+     * @SuppressWarnings("unchecked") Class<T> beanClass = (Class<T>)
+     * bean.getBeanClass();
+     * 
+     * return createDependentProvider(beanManager, beanClass, bean); }
+     * 
+     * private static <T> DependentProvider<T> createDependentProvider(
+     * BeanManager beanManager, Class<T> type, Bean<T> bean) {
+     * CreationalContext<T> cc = beanManager.createCreationalContext(bean);
+     * 
+     * @SuppressWarnings("unchecked") T instance = (T)
+     * beanManager.getReference(bean, type, cc);
+     * 
+     * return new DependentProvider<T>(bean, cc, instance); }
+     */
```

Comment:
Commented out method should be uncommented or removed.

Reply:
done